# Gráficos

## Objetivos:

Desarrollar el cuestionario para la evaluacion 3

In [7]:
import pandas as pd

df_1 = pd.read_excel("Base P3_C1.xlsx")

df_1

ID Categoría de Producto Nivel de Satisfacción
0      1              Juguetes            Satisfecho
1      2           Electrónica               Neutral
2      3                Libros               Neutral
3      4                Libros      Muy Insatisfecho
4      5                Libros          Insatisfecho
..   ...                   ...                   ...
495  496                  Ropa      Muy Insatisfecho
496  497                Libros          Insatisfecho
497  498                 Hogar            Satisfecho
498  499              Juguetes            Satisfecho
499  500                  Ropa          Insatisfecho

[500 rows x 3 columns]

Crear la tabla de frecuencias absolutas

In [9]:
tabla_abs = pd.crosstab(
    df_1["Categoría de Producto"],
    df_1["Nivel de Satisfacción"]
)
tabla_abs

Nivel de Satisfacción  Insatisfecho  Muy Insatisfecho  Muy Satisfecho  \
Categoría de Producto                                                   
Electrónica                      21                20              15   
Hogar                            14                19              17   
Juguetes                         20                17              21   
Libros                           22                26              22   
Ropa                             23                18              20   

Nivel de Satisfacción  Neutral  Satisfecho  
Categoría de Producto                       
Electrónica                 25          23  
Hogar                       20          18  
Juguetes                    16          17  
Libros                      30          21  
Ropa                        13          22

Crear la tabla de frecuencias relativas (respecto del total)

In [ ]:
tabla_rel = tabla_abs / tabla_abs.to_numpy().sum()
tabla_rel

Nivel de Satisfacción  Insatisfecho  Muy Insatisfecho  Muy Satisfecho  \
Categoría de Producto                                                   
Electrónica                   0.042             0.040           0.030   
Hogar                         0.028             0.038           0.034   
Juguetes                      0.040             0.034           0.042   
Libros                        0.044             0.052           0.044   
Ropa                          0.046             0.036           0.040   

Nivel de Satisfacción  Neutral  Satisfecho  
Categoría de Producto                       
Electrónica              0.050       0.046  
Hogar                    0.040       0.036  
Juguetes                 0.032       0.034  
Libros                   0.060       0.042  
Ropa                     0.026       0.044

In [16]:
## Que el cliente seleccionado esté Satisfecho con un producto de la categoría Juguetes
p1 = len(df_1[(df_1["Categoría de Producto"]=="Juguetes") & 
            (df_1["Nivel de Satisfacción"]=="Satisfecho")]) / len(df_1) * 100
round(p1, 2)

3.4

In [20]:
#Si se selecciona un cliente que está Muy satisfecho, que la categoría del producto sea Hogar.
ms = df_1[df_1["Nivel de Satisfacción"]=="Muy Satisfecho"]
p2 = len(ms[ms["Categoría de Producto"]=="Hogar"]) / len(ms) * 100
round(p2, 2)

17.89

In [23]:
#Si el cliente compró Ropa, ¿cuál es la probabilidad de que esté Insatisfecho?
ropa = df_1[df_1["Categoría de Producto"]=="Ropa"]
p3 = len(ropa[ropa["Nivel de Satisfacción"]=="Insatisfecho"]) / len(ropa) * 100
round(p3, 2)

23.96

✔️ **Verdadero**

La afirmación es correcta.

En Python, usando `scipy.stats.norm`, la función:

```python
norm.cdf(x, media, desviacion_estandar)
```

calcula justamente la **probabilidad acumulada** de que una variable aleatoria normal tome un valor **menor o igual** que *x*, es decir:

[
P(X \le x)
]

Y con esa misma función puedes obtener probabilidades dentro de un rango:

[
P(a \le X \le b) = \text{norm.cdf}(b) - \text{norm.cdf}(a)
]

Así que sí: **la afirmación es verdadera**.


📌 Recordatorio rápido de asimetría (skewness)

Asimetría > 0 → cola hacia la derecha (asimetría positiva).

Asimetría < 0 → cola hacia la izquierda (asimetría negativa).

Asimetría ≈ 0 → simétrica.

Cuando el problema da rangos:

Coeficiente de asimetría > 0,5 → fuertemente positivo

Coeficiente de asimetría < –0,5 → fuertemente negativo

–0,5 < asimetría < 0,5 → casi simétrica

📌 Interpretación de la curtosis

Curtosis > 0,5 → distribución leptocúrtica (pico muy alto y colas más pesadas).

Curtosis ≈ 0 → distribución mesocúrtica (similar a la normal).

Curtosis < –0,5 → distribución platicúrtica (pico más bajo y más ancha).

Resumen de cómo se obtuvieron los valores
Probabilidad	Tipo	Fórmula	Resultado
Satisfecho + Juguetes	Conjunta	17 / 500	3.4%
Hogar dado Muy Satisfecho	Condicional	17 / 95	17.9%
Insatisfecho dado Ropa	Condicional	23 / 96	24%

Vamos a calcular cada probabilidad usando la distribución normal con:

* **Media (μ) = 12 min**
* **Desviación estándar (σ) = 3 min**

Usamos valores típicos de la tabla Z.

---

# ✅ **1) Probabilidad de esperar como máximo 15 minutos**

[
P(X≤15)
]

Calculamos el puntaje Z:

[
Z=15−12/3​ = 1
]

De la tabla normal:

[
P(Z≤1)=0.8413
]

➡️ **84.1 %**

---

# ✅ **2) Probabilidad de esperar al menos 11 minutos**

[
P(X \ge 11)=1 - P(X < 11)
]

Puntaje Z:

[
Z=\frac{11-12}{3}=-0.3333
]

[
P(Z \le -0.33)=0.369
]

Entonces:

[
P(X \ge 11) = 1-0.369=0.631
]

➡️ **63.1 %**

---

# ✅ **3) Probabilidad de esperar entre 5 y 10 minutos**

[
P(5 \le X \le 10)=P(X \le 10)-P(X \le 5)
]

### Para 10 minutos:

[
Z=\frac{10-12}{3}=-0.6667
]
[
P(Z\le -0.67)=0.2525
]

### Para 5 minutos:

[
Z=\frac{5-12}{3}=-2.333
]
[
P(Z\le -2.33)=0.0099
]

[
P = 0.2525 - 0.0099 = 0.2426
]

➡️ **24.3 %**

---

# ✅ **RESPUESTAS FINALES (redondeadas a una cifra decimal)**

| Pregunta                                       | Resultado  |
| ---------------------------------------------- | ---------- |
| Probabilidad de esperar **como máximo 15 min** | **84.1 %** |
| Probabilidad de esperar **al menos 11 min**    | **63.1 %** |
| Probabilidad de esperar **entre 5 y 10 min**   | **24.3 %** |

---

Si quieres, puedo mostrarte cómo se calculan en Python usando `scipy`.


Sí, es posible que el peso siga (aprox.) una distribución normal.

Justificación:

En una distribución normal se cumple que media ≈ mediana ≈ moda.
Aquí: media = 75 kg, mediana = 75 kg, moda = 74 kg → son muy similares.

El coeficiente de asimetría = 0,2, que está cerca de 0, indica que la distribución es casi simétrica.

El coeficiente de curtosis = 0,1, también cercano a 0, sugiere que el “apuntamiento” es similar al de una normal (mesocúrtica).

Por estos valores (tendencias centrales casi iguales, asimetría y curtosis cercanas a 0), es razonable asumir que la variable peso podría aproximarse a una distribución normal.